WalletWasabi/Services/WalletService.cs

```diff
@@ -619,7 +619,7 @@ private bool ProcessTransaction(SmartTransaction tx)
 						TransactionCache.TryAdd(tx);
 
 						// If it's being mixed and anonset is not sufficient, then queue it.
-						if (newCoin.Unspent && ChaumianClient.HasIngredients && newCoin.Label.StartsWith("ZeroLink", StringComparison.Ordinal) && newCoin.AnonymitySet < ServiceConfiguration.MixUntilAnonymitySet)
+						if (newCoin.Unspent && ChaumianClient.HasIngredients && newCoin.AnonymitySet < ServiceConfiguration.MixUntilAnonymitySet && SpendsEnqueued(tx.Transaction))
```

Comment:
Consider passing the TxoRef[] instead of the transaction.
```c#
SpendsEnqueued(newCoin.SpentOutputs))
```

Reply:
Indeed. I needed the txid for something, but then I simplified it and the txid is not needed anymore, so I can just go with that.